src/utils/system_utils/tests/system_utils_ut.cpp

```diff
@@ -341,31 +344,30 @@ TEST_CASE_METHOD(TestCaseFixture, "SystemUtils_ForEachDir")
 
         REQUIRE(0 == ADUC_SystemUtils_MkDirRecursiveDefault(test_path.c_str()));
 
+        auto verify_dirs_fn = [&]()
         {
             std::vector<std::string> paths;
             REQUIRE(get_dirs_in_dir(TestPath(), paths) == 0);
             REQUIRE(paths.size() == 1);
             REQUIRE_THAT(paths[0], Equals(test_subdir));
-        }
+        };
+
+        verify_dirs_fn();
 
         int err = SystemUtils_ForEachDir(
             TestPath() /* baseDir */, nullptr /* excludeDir */, &functor /* perDirActionFunctor */);
         REQUIRE(err == 0);
 
-        CHECK_THAT(actual, Equals(expected));
+        std::sort(actual.begin(), actual.end(), std::less<std::string>());
```

Comment:
perhaps add a wrapper in this UT, `vector<string> SystemUtils_ForEachDir_Sorted()` which has a local "vector<string>"  calls SystemUtils_ForEachDir and then calls sort on the vector and returns it.  Would be a nice C++ wrapper for that method and would likely simplify some of the additions you've made here.

Reply:
as if your expected vector is already sorted, you should be able to use a single line:

`CHECK(ForEachDir_Sorted(.., ..) == vector({ "a", "b", "c" });`

Reply:
I've implemented it. The SystemUtils_ForEachDir takes a functor that has C-style callback function and context member as function object state.

Therefore, I created an static AddRecord method in the TestCaseFixture where it passes the context (which is the TestCaseFixture object instance) and the ForEachDirCallRecord object. 

The actual TEST_CASE_METHOD SECTION test code is now just 2-3 lines each.  Thanks for your suggestions!